### ISSUE TITLE
bootstrap-salt.sh fails to start salt-minion daemon on CentOS 6

### ISSUE THREAD

**Issue by tateeskew** [saltstack/salt-bootstrap#61]
As stated in the subject, bootstrap-salt.sh fails to start salt-minion daemon on CentOS 6.

I've tested this by using python-bootstrap and bootstrap-salt.

Below you can see the output of spinning up an instance.

```
[$] <> sudo salt-cloud -p cloudteam_aws_micro tatetest3
[INFO    ] Loaded configuration file: /etc/salt/cloud
[INFO    ] salt-cloud starting
[INFO    ] Creating Cloud VM tatetest3 in us-east-1
[INFO    ] Created node tatetest3
[WARNING ] Salt node waiting_for_ip 1
[WARNING ] Salt node waiting_for_ip 2
[WARNING ] Salt node waiting_for_ip 3
[WARNING ] Salt node waiting_for_ip 4
[WARNING ] Salt node waiting_for_ip 5
[WARNING ] Salt node waiting_for_ip 6
[WARNING ] Salt node waiting_for_ip 7
[WARNING ] Salt node waiting_for_ip 8
[WARNING ] Salt node waiting_for_ip 9
[WARNING ] Salt node waiting_for_ip 10
[WARNING ] Salt node waiting_for_ip 11
[WARNING ] Salt node waiting_for_ip 12
[WARNING ] Salt node waiting_for_ip 13
[WARNING ] Salt node waiting_for_ip 14
[WARNING ] Salt node waiting_for_ip 15
[WARNING ] Salt node waiting_for_ip 16
[WARNING ] Salt node waiting_for_ip 17
[WARNING ] Salt node waiting_for_ip 18
[WARNING ] Salt node waiting_for_ip 19
[WARNING ] Salt node waiting_for_ip 20
[WARNING ] Salt node waiting_for_ip 21
[WARNING ] Salt node waiting_for_ip 22
[WARNING ] Salt node waiting_for_ip 23
[INFO    ] Salt node data. Public_ip: 75.101.252.103
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
Tue Feb 26 18:08:55 UTC 2013
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
Tue Feb 26 18:08:55 UTC 2013
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
tmpNlRmP3                                                                                                                                                                                                   100% 1850     1.8KB/s   00:00    
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
Connection to 75.101.252.103 closed.
tmpzx3igq                                                                                                                                                                                                   100%  796     0.8KB/s   00:00    
tmpNQf9E8                                                                                                                                                                                                   100%   82     0.1KB/s   00:00    
tmpetmoaL                                                                                                                                                                                                   100%   57KB  56.8KB/s   00:00    
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
Connection to 75.101.252.103 closed.
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
 *  INFO: /bin/sh /tmp/deploy.sh -- Version 1.4

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   2.6.32-279.22.1.el6.x86_64
 *  INFO:   Distribution: CentOS 6.3

 *  INFO: Installing minion
 *  INFO: Found function install_centos_stable_deps
 *  INFO: Found function config_salt
 *  INFO: Found function install_centos_stable
 *  INFO: Found function install_centos_stable_post
 *  INFO: Found function install_centos_start_daemons
 *  INFO: Running install_centos_stable_deps()
Retrieving http://mirrors.kernel.org/fedora-epel/6/x86_64/epel-release-6-8.noarch.rpm
Preparing...                ##################################################
epel-release                ##################################################
Loaded plugins: fastestmirror, presto
Determining fastest mirrors
 * base: mirror.liberty.edu
 * epel: mirror.symnds.com
 * extras: mirror.symnds.com
 * updates: mirror.ash.fastserv.com
Setting up Update Process
Resolving Dependencies
--> Running transaction check
---> Package libproxy.x86_64 0:0.3.0-3.el6_3 will be updated
---> Package libproxy.x86_64 0:0.3.0-4.el6_3 will be an update
---> Package libproxy-bin.x86_64 0:0.3.0-3.el6_3 will be updated
---> Package libproxy-bin.x86_64 0:0.3.0-4.el6_3 will be an update
---> Package libproxy-python.x86_64 0:0.3.0-3.el6_3 will be updated
---> Package libproxy-python.x86_64 0:0.3.0-4.el6_3 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package                Arch          Version              Repository      Size
================================================================================
Updating:
 libproxy               x86_64        0.3.0-4.el6_3        updates         39 k
 libproxy-bin           x86_64        0.3.0-4.el6_3        updates        8.2 k
 libproxy-python        x86_64        0.3.0-4.el6_3        updates        8.4 k

Transaction Summary
================================================================================
Upgrade       3 Package(s)

Total download size: 55 k
Downloading Packages:
Setting up and reading Presto delta metadata
Processing delta metadata
Download delta size: 15 k
Presto reduced the update size by 68% (from 47 k to 15 k).
Package(s) data still to download: 8.2 k
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Updating   : libproxy-python-0.3.0-4.el6_3.x86_64                         1/6 
  Updating   : libproxy-0.3.0-4.el6_3.x86_64                                2/6 
  Updating   : libproxy-bin-0.3.0-4.el6_3.x86_64                            3/6 
  Cleanup    : libproxy-python-0.3.0-3.el6_3.x86_64                         4/6 
  Cleanup    : libproxy-0.3.0-3.el6_3.x86_64                                5/6 
  Cleanup    : libproxy-bin-0.3.0-3.el6_3.x86_64                            6/6 
  Verifying  : libproxy-bin-0.3.0-4.el6_3.x86_64                            1/6 
  Verifying  : libproxy-python-0.3.0-4.el6_3.x86_64                         2/6 
  Verifying  : libproxy-0.3.0-4.el6_3.x86_64                                3/6 
  Verifying  : libproxy-python-0.3.0-3.el6_3.x86_64                         4/6 
  Verifying  : libproxy-bin-0.3.0-3.el6_3.x86_64                            5/6 
  Verifying  : libproxy-0.3.0-3.el6_3.x86_64                                6/6 

Updated:
  libproxy.x86_64 0:0.3.0-4.el6_3          libproxy-bin.x86_64 0:0.3.0-4.el6_3  
  libproxy-python.x86_64 0:0.3.0-4.el6_3  

Complete!
 *  INFO: Running config_salt()
 *  INFO: Running install_centos_stable()
Loaded plugins: fastestmirror, presto
Loading mirror speeds from cached hostfile
 * base: mirror.liberty.edu
 * epel: mirror.symnds.com
 * epel-testing: mirror.symnds.com
 * extras: mirror.symnds.com
 * updates: mirror.ash.fastserv.com
Setting up Install Process
Resolving Dependencies
--> Running transaction check
---> Package salt-minion.noarch 0:0.13.1-1.el6 will be installed
--> Processing Dependency: salt = 0.13.1-1.el6 for package: salt-minion-0.13.1-1.el6.noarch
--> Running transaction check
---> Package salt.noarch 0:0.13.1-1.el6 will be installed
--> Processing Dependency: python-zmq for package: salt-0.13.1-1.el6.noarch
--> Processing Dependency: python-msgpack for package: salt-0.13.1-1.el6.noarch
--> Processing Dependency: python-jinja2 for package: salt-0.13.1-1.el6.noarch
--> Processing Dependency: python-crypto for package: salt-0.13.1-1.el6.noarch
--> Processing Dependency: m2crypto for package: salt-0.13.1-1.el6.noarch
--> Processing Dependency: PyYAML for package: salt-0.13.1-1.el6.noarch
--> Running transaction check
---> Package PyYAML.x86_64 0:3.10-3.el6 will be installed
--> Processing Dependency: libyaml-0.so.2()(64bit) for package: PyYAML-3.10-3.el6.x86_64
---> Package m2crypto.x86_64 0:0.20.2-9.el6 will be installed
---> Package python-crypto.x86_64 0:2.0.1-22.el6 will be installed
---> Package python-jinja2.x86_64 0:2.2.1-1.el6 will be installed
--> Processing Dependency: python-babel >= 0.8 for package: python-jinja2-2.2.1-1.el6.x86_64
---> Package python-msgpack.x86_64 0:0.1.9-2.el6 will be installed
---> Package python-zmq.x86_64 0:2.2.0.1-1.el6 will be installed
--> Processing Dependency: libzmq.so.3()(64bit) for package: python-zmq-2.2.0.1-1.el6.x86_64
--> Running transaction check
---> Package libyaml.x86_64 0:0.1.3-1.el6 will be installed
---> Package python-babel.noarch 0:0.9.4-5.1.el6 will be installed
---> Package zeromq3.x86_64 0:3.2.2-2.el6 will be installed
--> Processing Dependency: libpgm-5.1.so.0()(64bit) for package: zeromq3-3.2.2-2.el6.x86_64
--> Running transaction check
---> Package openpgm.x86_64 0:5.1.118-3.el6 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package              Arch         Version             Repository          Size
================================================================================
Installing:
 salt-minion          noarch       0.13.1-1.el6        epel-testing        17 k
Installing for dependencies:
 PyYAML               x86_64       3.10-3.el6          epel               157 k
 libyaml              x86_64       0.1.3-1.el6         epel                52 k
 m2crypto             x86_64       0.20.2-9.el6        base               471 k
 openpgm              x86_64       5.1.118-3.el6       epel               165 k
 python-babel         noarch       0.9.4-5.1.el6       base               1.4 M
 python-crypto        x86_64       2.0.1-22.el6        base               159 k
 python-jinja2        x86_64       2.2.1-1.el6         base               465 k
 python-msgpack       x86_64       0.1.9-2.el6         epel                28 k
 python-zmq           x86_64       2.2.0.1-1.el6       epel               424 k
 salt                 noarch       0.13.1-1.el6        epel-testing       1.3 M
 zeromq3              x86_64       3.2.2-2.el6         epel               326 k

Transaction Summary
================================================================================
Install      12 Package(s)

Total download size: 4.9 M
Installed size: 18 M
Downloading Packages:
Setting up and reading Presto delta metadata
Processing delta metadata
Package(s) data still to download: 4.9 M
--------------------------------------------------------------------------------
Total                                           1.9 MB/s | 4.9 MB     00:02     
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Installing : libyaml-0.1.3-1.el6.x86_64                                  1/12 
  Installing : PyYAML-3.10-3.el6.x86_64                                    2/12 
  Installing : python-crypto-2.0.1-22.el6.x86_64                           3/12 
  Installing : m2crypto-0.20.2-9.el6.x86_64                                4/12 
  Installing : python-msgpack-0.1.9-2.el6.x86_64                           5/12 
  Installing : python-babel-0.9.4-5.1.el6.noarch                           6/12 
  Installing : python-jinja2-2.2.1-1.el6.x86_64                            7/12 
  Installing : openpgm-5.1.118-3.el6.x86_64                                8/12 
  Installing : zeromq3-3.2.2-2.el6.x86_64                                  9/12 
  Installing : python-zmq-2.2.0.1-1.el6.x86_64                            10/12 
  Installing : salt-0.13.1-1.el6.noarch                                   11/12 
  Installing : salt-minion-0.13.1-1.el6.noarch                            12/12 
warning: /etc/salt/minion created as /etc/salt/minion.rpmnew
  Verifying  : openpgm-5.1.118-3.el6.x86_64                                1/12 
  Verifying  : python-babel-0.9.4-5.1.el6.noarch                           2/12 
  Verifying  : python-msgpack-0.1.9-2.el6.x86_64                           3/12 
  Verifying  : python-jinja2-2.2.1-1.el6.x86_64                            4/12 
  Verifying  : m2crypto-0.20.2-9.el6.x86_64                                5/12 
  Verifying  : zeromq3-3.2.2-2.el6.x86_64                                  6/12 
  Verifying  : python-zmq-2.2.0.1-1.el6.x86_64                             7/12 
  Verifying  : PyYAML-3.10-3.el6.x86_64                                    8/12 
  Verifying  : salt-minion-0.13.1-1.el6.noarch                             9/12 
  Verifying  : python-crypto-2.0.1-22.el6.x86_64                          10/12 
  Verifying  : libyaml-0.1.3-1.el6.x86_64                                 11/12 
  Verifying  : salt-0.13.1-1.el6.noarch                                   12/12 

Installed:
  salt-minion.noarch 0:0.13.1-1.el6                                             

Dependency Installed:
  PyYAML.x86_64 0:3.10-3.el6             libyaml.x86_64 0:0.1.3-1.el6          
  m2crypto.x86_64 0:0.20.2-9.el6         openpgm.x86_64 0:5.1.118-3.el6        
  python-babel.noarch 0:0.9.4-5.1.el6    python-crypto.x86_64 0:2.0.1-22.el6   
  python-jinja2.x86_64 0:2.2.1-1.el6     python-msgpack.x86_64 0:0.1.9-2.el6   
  python-zmq.x86_64 0:2.2.0.1-1.el6      salt.noarch 0:0.13.1-1.el6            
  zeromq3.x86_64 0:3.2.2-2.el6          

Complete!
 *  INFO: Running install_centos_stable_post()
 *  INFO: Running install_centos_start_daemons()
 *  INFO: Salt installed!
Starting salt-minion daemon: Connection to 75.101.252.103 closed.
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
Connection to 75.101.252.103 closed.
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
rm: cannot remove `/tmp/minion.pub': No such file or directory
Connection to 75.101.252.103 closed.
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
rm: cannot remove `/tmp/minion.pem': No such file or directory
Connection to 75.101.252.103 closed.
Warning: Permanently added '75.101.252.103' (RSA) to the list of known hosts.
rm: cannot remove `/tmp/minion': No such file or directory
Connection to 75.101.252.103 closed.
[INFO    ] Salt installed on tatetest3
[INFO    ] Created Cloud VM tatetest3 with the following values:
[INFO    ]   private_ips: ['10.117.85.157']
[INFO    ]   extra: {'status': 'running', 'productcode': [], 'groups': [None, None], 'tags': {'Name': 'tatetest3'}, 'instanceId': 'i-8c28bcff', 'dns_name': 'ec2-75-101-252-103.compute-1.amazonaws.com', 'launchdatetime': '2013-02-26T18:07:59.000Z', 'imageId': 'ami-ae78e9c7', 'kernelid': 'aki-88aa75e1', 'keyname': 'coreservice', 'availability': 'us-east-1a', 'clienttoken': '', 'launchindex': '0', 'ramdiskid': None, 'private_dns': 'ip-10-117-85-157.ec2.internal', 'instancetype': 't1.micro'}
[INFO    ]   image: None
[INFO    ]   _uuid: None
[INFO    ]   driver: <libcloud.compute.drivers.ec2.EC2NodeDriver object at 0x2dfa310>
[INFO    ]   state: 0
[INFO    ]   public_ips: ['75.101.252.103']
[INFO    ]   size: None
[INFO    ]   id: i-8c28bcff
[INFO    ]   name: tatetest3
```

**Comment by s0undt3ch** [comment]
Thanks @tateeskew! We'll have a look at this.

PS: Updated the ticket details so we can have a better formatting on the output.

**Comment by s0undt3ch** [comment]
Fixed by da57d3627744fca1a97375106519cb90ef9ca898 on #62
